contracts/NestedFactory.sol

```diff
@@ -127,34 +124,52 @@ contract NestedFactory is INestedFactory, ReentrancyGuard, Ownable, MixinOperato
     /// @inheritdoc INestedFactory
     function addTokens(
         uint256 _nftId,
-        IERC20 _sellToken,
-        uint256 _sellTokenAmount,
-        Order[] calldata _orders
+        BatchedOrder calldata _batch
     ) external payable override nonReentrant onlyTokenOwner(_nftId) {
-        require(_orders.length != 0, "NF: INVALID_ORDERS");
         require(nestedRecords.getAssetReserve(_nftId) == address(reserve), "NF: RESERVE_MISMATCH");
 
-        (uint256 fees, IERC20 tokenSold) = _submitInOrders(_nftId, _sellToken, _sellTokenAmount, _orders, true, false);
+        (uint256 fees, IERC20 tokenSold) = _submitInOrders(_nftId, _batch, true, false);
         _transferFeeWithRoyalty(fees, tokenSold, _nftId);
         emit NftUpdated(_nftId);
     }
 
     /// @inheritdoc INestedFactory
     function swapTokenForTokens(
         uint256 _nftId,
-        IERC20 _sellToken,
-        uint256 _sellTokenAmount,
-        Order[] calldata _orders
+        BatchedOrder calldata _batch
     ) external override nonReentrant onlyTokenOwner(_nftId) isUnlocked(_nftId) {
-        require(_orders.length != 0, "NF: INVALID_ORDERS");
         require(nestedRecords.getAssetReserve(_nftId) == address(reserve), "NF: RESERVE_MISMATCH");
 
-        (uint256 fees, IERC20 tokenSold) = _submitInOrders(_nftId, _sellToken, _sellTokenAmount, _orders, true, true);
+        (uint256 fees, IERC20 tokenSold) = _submitInOrders(_nftId, _batch, true, true);
         _transferFeeWithRoyalty(fees, tokenSold, _nftId);
 
         emit NftUpdated(_nftId);
     }
 
+    /// @inheritdoc INestedFactory
+    function swapTokensForTokens(uint256 _nftId, BatchedOrder[] calldata _batchedOrders)
+        external
+        override
+        nonReentrant
+        onlyTokenOwner(_nftId)
+        isUnlocked(_nftId)
+    {
+        require(_batchedOrders.length != 0, "NF: INVALID_MULTI_ORDERS");
+        require(nestedRecords.getAssetReserve(_nftId) == address(reserve), "NF: RESERVE_MISMATCH");
+
+        for (uint256 i = 0; i < _batchedOrders.length; i++) {
+            (uint256 fees, IERC20 tokenSold) = _submitInOrders(
+                _nftId,
+                _batchedOrders[i],
+                true,
+                true
+            );
+            _transferFeeWithRoyalty(fees, tokenSold, _nftId);
```

Comment:
❔ ⚠️  I discussed with Rudy that we lower the fees for multiswaps, to make up for the increased gas cost and also not penalize users who regularly follow the moves of the people they copy.

Let's discuss it with Rudy. In that case, we can add a duplicate function of `_transferFeeWithRoyalty`

Reply:
Maybe by given the fees amount as a parameter to override the actual percentage?
This is always 1% (hardcoded) :
```javascript
uint256 feeAmount = _amount / 100;
```

Reply:
I think it'd be more efficient to duplicate this function so that an extra parameter isn't passed for every call made to the factory